osu.Game/Beatmaps/Formats/LegacyBeatmapDecoder.cs

```diff
@@ -11,12 +11,15 @@
 using osu.Game.Beatmaps.Legacy;
 using osu.Game.Beatmaps.Timing;
 using osu.Game.IO;
+using osu.Game.Rulesets;
 using osu.Game.Rulesets.Objects.Legacy;
 
 namespace osu.Game.Beatmaps.Formats
 {
     public class LegacyBeatmapDecoder : LegacyDecoder<Beatmap>
     {
+        protected internal static RulesetStore RulesetStore;
```

Comment:
`protected internal` makes it so anything in the same assembly, and any deriver. can potentially modify this to its heart's content bypassing `RegisterDependencies()` entirely. I think `internal` would be enough for this usage, and it would prevent outside consumers from inheriting and shooting themselves in the foot by messing with this. `private protected` would be even better, but it's not possible to do that without moving the ruleset store out to the abstract `Decoder` too, so...

I suppose it's not a huge worry because any other mutation should by all means be caught at review time but I dunno, always would be good to have the compiler enforce this.